tests/draft2019-09/ref.json

```diff
@@ -357,5 +357,28 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "ref creates new scope when adjacent to keywords",
+        "schema": {
+            "$defs": {
+                "A": {
+                    "unevaluatedProperties": false
+                }
+            },
+            "properties": {
+                "prop1": {
+                    "type": "string"
+                }
+            },
+            "$ref": "#/$defs/A"
+        },
+        "tests": [
+            {
+                "prop1": "match",
+                "prop2": 1,
```

Comment:
If `$ref` is resolved as a literal in-place substitution, then `prop2` would be disallowed.

Reply:
wouldn't this mean both prop1 and prop2 are disallowed since unevaluatedProps cannot see either of them?

Reply:
It would, yes.

Reply:
since the properties are disallowed, I'm confused by the `valid: true` expected result

Reply:
It's valid because `unevaluatedProperties` is in its own scope and can't see.... now I'm confused.  Hang on.  Will edit.

Reply:
The test set up wrong anyway.

Reply:
okay.  fixed.